### ISSUE TITLE
Natural Keys filter is not working

### ISSUE THREAD

**Issue by pol51** [biosustain/potion#75]
I tried to use natural keys but it doesn't seems to work like described in the quickstart.
I also tried to follow the quickstart guide with "quickstart_relation.py" from v0.12.3 and I get this error:

```
$ http GET :5000/book where=='{"author": ["Charles", "Darwin"]}'
HTTP/1.0 400 BAD REQUEST
Content-Length: 2333
Content-Type: application/json
Date: Tue, 12 Apr 2016 12:22:42 GMT
Server: Werkzeug/0.11.5 Python/2.7.11

{
    "errors": [
        {
            "path": [
                "where", 
                "author"
            ], 
            "validationOf": {
                "anyOf": [
                    {
                        "additionalProperties": false, 
                        "properties": {
                            "$ref": {
                                "format": "uri", 
                                "pattern": "^\\/author\\/[^/]+$", 
                                "type": "string"
                            }
                        }, 
                        "type": "object"
                    }, 
                    {
                        "additionalProperties": false, 
                        "properties": {
                            "$eq": {
                                "additionalProperties": false, 
                                "properties": {
                                    "$ref": {
                                        "format": "uri", 
                                        "pattern": "^\\/author\\/[^/]+$", 
                                        "type": "string"
                                    }
                                }, 
                                "type": "object"
                            }
                        }, 
                        "required": [
                            "$eq"
                        ], 
                        "type": "object"
                    }, 
                    {
                        "additionalProperties": false, 
                        "properties": {
                            "$ne": {
                                "additionalProperties": false, 
                                "properties": {
                                    "$ref": {
                                        "format": "uri", 
                                        "pattern": "^\\/author\\/[^/]+$", 
                                        "type": "string"
                                    }
                                }, 
                                "type": "object"
                            }
                        }, 
                        "required": [
                            "$ne"
                        ], 
                        "type": "object"
                    }, 
                    {
                        "additionalProperties": false, 
                        "properties": {
                            "$in": {
                                "items": {
                                    "additionalProperties": false, 
                                    "properties": {
                                        "$ref": {
                                            "format": "uri", 
                                            "pattern": "^\\/author\\/[^/]+$", 
                                            "type": "string"
                                        }
                                    }, 
                                    "type": "object"
                                }, 
                                "minItems": 0, 
                                "type": "array", 
                                "uniqueItems": true
                            }
                        }, 
                        "required": [
                            "$in"
                        ], 
                        "type": "object"
                    }
                ]
            }
        }
    ], 
    "message": "Bad Request", 
    "status": 400
}
```

**Comment by lyschoening** [comment]
You're right, it seems to be broken for ?where= queries (still works for create/update). I will have a look at why that is.

**Comment by lyschoening** [comment]
Fixed in v0.12.4
